examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1344 @@
+    const QGauss<1>       quadrature_1D;
+
+    // We need three separate FE spaces.
+    // The first manages the TraceFEM space which is active on intersected
+    // element. The second manages the discrete
+    // level set function that describes the geometry of the surface. The third
+    // handles a piecewise constant space which is used solely for the
+    // accumulation of refinement indicators and of errors.
+    hp::FECollection<dim> fe_collection;
+    const FE_Q<dim>       level_set_fe;
+
+    // The corresponding DOF-handlers are given by
+    DoFHandler<dim> dof_handler;
+    DoFHandler<dim> level_set_dof_handler;
+
+    // Since we will adaptively refine the bulk triangulation, two constraints
+    // are needed: one for the solution space and another for the level-set
+    // space.
+    AffineConstraints<double> constraints;
+    AffineConstraints<double> level_set_constraints;
+
+    // Discrete vectors initialized with dof_handler and level_set_dof_handler.
+    VectorType    completely_distributed_solution;
+    VectorType    locally_relevant_solution;
+    VectorType    locally_relevant_exact;
+    VectorType    level_set;
+    Vector<float> activeFE_indicator;
+
+    // Mesh_classifier separates intersected elements and non-intersected ones
+    // in the fe_collection
+    NonMatching::MeshClassifier<dim> mesh_classifier;
+    const MappingQ1<dim>             mapping;
+
+    // Any TraceFEM need a stabilization, and we choose the normal-gradient,
+    // volume stabilization.
+    NormalGradientVolumeStabilization<dim> stabilization_scheme;
+
+    // Discrete right-hand side and the final matrix correspondent to
+    // dof_handler.
+    VectorType global_rhs;
+    MatrixType global_matrix;
+
+    SparsityPattern sparsity_pattern;
+
+    IndexSet locally_owned_dofs;
+    IndexSet locally_relevant_dofs;
+
+    // Depending on the type of the quadrature, surface, face or volume, we need
+    // to define different update flags.
+    NonMatching::RegionUpdateFlags surface_update_flags;
+    UpdateFlags                    face_update_flags;
+    UpdateFlags                    cell_update_flags;
+
+    // The following displays the errors of the convergence test.
+    ConvergenceTable convergence_table;
+    unsigned int     number_of_iterations;
+    double           average;
+    double           area;
+    double           error_L2_sqr;
+    double           error_H1_sqr;
+    double           error_stab_sqr;
+
+    int number_of_proc;
+  };
+
+  // @sect2{The constructor of the solver}
+  // One may adjust the parameters of TraceFEM here.
+  template <int dim>
+  LaplaceBeltramiSolver<dim>::LaplaceBeltramiSolver()
+    : mpi_communicator(MPI_COMM_WORLD)
+    , triangulation(
+        mpi_communicator,
+        typename Triangulation<dim>::MeshSmoothing(
+          Triangulation<dim>::
+            none //smoothing_on_refinement |  Triangulation<dim>::smoothing_on_coarsening
+          ),
```

Comment:
If you don't need the smoothing, just remove what you have as a comment here. Then you can probably also drop the whole argument to the constructor.